api/envoy/extensions/request_id/uuid/v3/uuid.proto

```diff
@@ -40,4 +55,7 @@ message UuidRequestIdConfig {
   // stable sampling of traces, access logs, etc. will no longer work and only random sampling will
   // be possible.
   google.protobuf.BoolValue pack_trace_reason = 1;
+
+  // Sets the policy for mutation related to :ref:`x-request-id<config_http_conn_man_headers_x-request-id>`.
+  RequestIDMutationPolicy mutation_policy = 2;
```

Comment:
Can we just make this a bool value `use_request_id_for_trace_sampling` and default it to true? And then have relevant docs and code/APIs match? I think that will be simpler to understand.

Reply:
We discussed this somewhat in https://github.com/envoyproxy/envoy/pull/16770#discussion_r647909255. Do you think the rationale for enum there is relatively weak?

Reply:
IMO a bool is OK, but if we want to keep an enum, I would probably name it something else. I don't think "mutation policy" makes sense. I would call it `RequestIdTraceSamplingPolicy`. But once we do that, will there ever really be more than 2 values?

Reply:
`SKIP` is just a "policy" on how to change the value depending on the request ID. So it is possible that a new policy will be added in the future.

Reply:
If we think there will be actual policies I would not use an enum as that is pretty fragile. I would use a one of with either bools or empty messages inside it. I would also rename.

Reply:
+1 to make it BoolValue with default true. I don't think there will be more.

Reply:
FWIW, I don't feel that strongly here. My instinct as API shepherd is to ask for bools whether we are best served by enum/oneof, since this is a frequent place we paint outselves into a corner. In this case I think we can arguably just go with the more direct bool.